UndertaleModTool/Editors/UndertaleGameObjectEditor.xaml.cs

```diff
@@ -68,5 +68,33 @@ private void ComboBox_DropDownClosed(object sender, EventArgs e)
         {
             handleMouseScroll = true;
         }
+        private void UndertaleObjectReference_Loaded(object sender, RoutedEventArgs e)
+        {
+            var objRef = sender as UndertaleObjectReference;
+
+            objRef.ClearRemoveClickHandler();
+            objRef.RemoveButton.Click += Remove_Click_Override;
+            objRef.RemoveButton.ToolTip = "Remove action";
+            objRef.RemoveButton.IsEnabled = true;
+        }
+        private void Remove_Click_Override(object sender, RoutedEventArgs e)
+        {
+            var btn = (ButtonDark)sender;
+            var objRef = (UndertaleObjectReference)((Grid)btn.Parent).Parent;
+
+            var obj = (UndertaleGameObject)DataContext;
+            var evType = objRef.ObjectEventType;
+            var evSubtype = objRef.ObjectEventSubtype;
+            var action = (UndertaleGameObject.EventAction)btn.DataContext;
+            var evList = ((UndertaleGameObject)DataContext).Events[(int)evType];
+
+            var ev = evList.FirstOrDefault(x => x.EventSubtype == evSubtype);
+            if (ev is null) return;
+            ev.Actions.Remove(action);
+            if (ev.Actions.Count <= 0)
```

Comment:
Hmm... Not sure how I feel about this behavior, or maybe it's slightly broken right now. It seems currently you have to click the remove button twice in order to remove an action entirely (the first click seems to just detach the code entry, weirdly).

Reply:
it used to be like that, but i thought i fixed that

Reply:
yea it does remove the action in one click for me

Reply:
Oh, it looks like I accidentally pulled an earlier version. My bad!! Looks like it's working great on the latest version.